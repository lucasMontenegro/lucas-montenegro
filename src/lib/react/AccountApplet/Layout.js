import { makeStyles } from "@material-ui/core/styles"
import React, { Fragment } from "react"
import Typography from "@material-ui/core/Typography"
const useStyles = makeStyles(theme => ({
  root: {
    margin: theme.spacing(1),
    display: `flex`,
  },
  avatar: {
    height: theme.spacing(8),
    width: theme.spacing(8),
    margin: theme.spacing(1),
    borderRadius: `50%`,
    overflow: `hidden`,
  },
  text: {
    height: theme.spacing(8),
    margin: theme.spacing(1),
    display: `flex`,
    flexDirection: `column`,
    justifyContent: `center`,
  },
  topText: {
    marginBottom: theme.spacing(0.75),
  },
}), { name: `lib-react-account_applet-layout` })
export default function Layout (props) {
  const classes = useStyles()
  const { text } = props
  return (
    <div className={classes.root}>
      <div className={classes.avatar}>{props.avatar}</div>
      <div className={classes.text}>
        {
          text ? <div>{text}</div> :
          (
            <Fragment>
              <Typography className={classes.topText} noWrap component="div" variant="body1">
                {props.topText}
              </Typography>
              <div>{props.bottomText}</div>
            </Fragment>
          )
        }
      </div>
    </div>
  )
}