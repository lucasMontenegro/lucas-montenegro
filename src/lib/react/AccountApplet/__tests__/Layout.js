import renderer from "react-test-renderer"
import React from "react"
import Layout from "../Layout"
jest.mock(`@material-ui/core/styles`, () => ({
  __esModule: true,
  makeStyles: (styles, options) => () => {
    const classes = styles({ spacing: n => 8*n })
    return Object.keys(classes).reduce((obj, key) => {
      obj[key] = `${options.name}-${key}(${JSON.stringify(classes[key]).replace(/"/gm, `'`)})`
      return obj
    }, {})
  },
}))
jest.mock(`@material-ui/core/Typography`, () => {
  const React = jest.requireActual("react")
  return {
    __esModule: true,
    default: props => <div {...props} className={`Typography ${props.className}`} />,
  }
})
describe(`../Layout`, () => {
  jestUtils.describeDependencies({
    deps: [
      `@material-ui/core`,
      `react`,
      `react-test-renderer`,
    ],
    relativeBasePath: __dirname,
  })
  describe.each([[`props.text`], [undefined]])(`<Layout /> (text %j)`, text => {
    it(`should render`, () => {
      expect(renderer.create(
        <Layout
          avatar="props.avatar"
          text={text}
          topText="props.topText"
          bottomText="props.bottomText"
        />
      )).toMatchSnapshot()
    })
  })
})