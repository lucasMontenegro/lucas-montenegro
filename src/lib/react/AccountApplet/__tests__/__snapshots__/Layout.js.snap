// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`../Layout <Layout /> (text "props.text") should render 1`] = `
<div
  className="lib-react-account_applet-layout-root({'margin':8,'display':'flex'})"
>
  <div
    className="lib-react-account_applet-layout-avatar({'height':64,'width':64,'margin':8,'borderRadius':'50%','overflow':'hidden'})"
  >
    props.avatar
  </div>
  <div
    className="lib-react-account_applet-layout-text({'height':64,'margin':8,'display':'flex','flexDirection':'column','justifyContent':'center'})"
  >
    <div>
      props.text
    </div>
  </div>
</div>
`;

exports[`../Layout <Layout /> (text undefined) should render 1`] = `
<div
  className="lib-react-account_applet-layout-root({'margin':8,'display':'flex'})"
>
  <div
    className="lib-react-account_applet-layout-avatar({'height':64,'width':64,'margin':8,'borderRadius':'50%','overflow':'hidden'})"
  >
    props.avatar
  </div>
  <div
    className="lib-react-account_applet-layout-text({'height':64,'margin':8,'display':'flex','flexDirection':'column','justifyContent':'center'})"
  >
    <div
      className="Typography lib-react-account_applet-layout-topText({'marginBottom':6})"
      component="div"
      noWrap={true}
      variant="body1"
    >
      props.topText
    </div>
    <div>
      props.bottomText
    </div>
  </div>
</div>
`;

exports[`../Layout external dependencies @material-ui/core should match the installed version 1`] = `"^4.5.0"`;

exports[`../Layout external dependencies react should match the installed version 1`] = `"^16.10.1"`;

exports[`../Layout external dependencies react-test-renderer should match the installed version 1`] = `"^16.10.1"`;
